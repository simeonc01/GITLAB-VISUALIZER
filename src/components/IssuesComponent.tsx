import React, { useContext, useEffect, useState } from "react";
import { Issue } from "../util/types";
import { GitLabContext } from "./GitlabProvider";
import Box from "@mui/material/Box";
import Collapse from "@mui/material/Collapse";
import IconButton from "@mui/material/IconButton";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Typography from "@mui/material/Typography";
import Paper from "@mui/material/Paper";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
import { Description } from "@mui/icons-material";

function IssuesComponent() {
  const [issues, setIssues] = useState<Issue[]>([]);

  const context = useContext(GitLabContext);

  useEffect(() => {
    const tempIssues = context.issues;
    if (tempIssues !== null) setIssues(tempIssues);
    console.log(issues);
  }, [context.issues]);

  function createData(
    title: string,
    assignee: string,
    created: Date,
    price: number,
    description: string
  ) {
    return {
      title,
      assignee,
      created,
      price,
      description,
      history: [
        {
          description: "2020-01-02",
          customerId: "11091700",
          amount: 3,
        },
        {
          date: "2020-01-02",
          customerId: "Anonymous",
          amount: 1,
        },
      ],
    };
  }

  const rows: ReturnType<typeof createData>[] = issues.map((issue) => {
    return createData(
      issue.title,
      "Navn",
      issue.created_at,
      3.3,
      issue.description
    );
  });

  function Row(props: { row: ReturnType<typeof createData> }) {
    const { row } = props;
    const [open, setOpen] = React.useState(false);

    return (
      <React.Fragment>
        <TableRow sx={{ "& > *": { borderBottom: "unset" } }}>
          <TableCell>
            <IconButton
              aria-label="expand row"
              size="small"
              onClick={() => setOpen(!open)}
            >
              {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
            </IconButton>
          </TableCell>
          <TableCell component="th" scope="row">
            {row.title}
          </TableCell>
          <TableCell>{row.assignee}</TableCell>
          <TableCell align="right">{row.created.toString()}</TableCell>
        </TableRow>
        <TableRow>
          <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
            <Collapse in={open} timeout="auto" unmountOnExit>
              <Box sx={{ margin: 1 }}>
                <Typography variant="h6" gutterBottom component="div">
                  Description
                </Typography>
                <Table size="small" aria-label="purchases">
                  <TableBody>
                    <TableRow>
                      <TableCell>{row.description}</TableCell>
                    </TableRow>
                  </TableBody>
                </Table>
              </Box>
            </Collapse>
          </TableCell>
        </TableRow>
      </React.Fragment>
    );
  }

  return (
    <div>
      <h1>Issues Component</h1>
      <TableContainer component={Paper}>
        <Table aria-label="Commits-table">
          <TableHead>
            <TableRow>
              <TableCell />
              <TableCell>Issue</TableCell>
              <TableCell>Author</TableCell>
              <TableCell align="right">Date</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map((row) => (
              <Row key={row.title} row={row} />
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}

export default IssuesComponent;
