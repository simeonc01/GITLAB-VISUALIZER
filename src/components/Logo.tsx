import React from "react";

export default function Logo() {
  return (
    <svg width="55" height="55">
      <circle
        cx="25"
        cy="25"
        r="25"
        fill="grey"
        stroke="white"
        stroke-width="2px"
        mask="url(#quarter)"
      />
      <line x1="25" y1="0" x2="25" y2="25" stroke="white" stroke-width="2px" />
      <line x1="25" y1="25" x2="50" y2="25" stroke="white" stroke-width="2px" />
      <line
        x1="25"
        y1="25"
        x2="42.5"
        y2="7.5"
        stroke="white"
        stroke-width="2px"
      />
      <defs>
        <mask id="quarter">
          <rect x="0" y="0" width="25" height="50" fill="white" />
          <rect x="0" y="25" width="50" height="25" fill="white" />
        </mask>
      </defs>
    </svg>
  );
}
