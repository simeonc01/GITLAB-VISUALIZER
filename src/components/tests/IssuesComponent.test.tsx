import { expect, test } from "@jest/globals";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import {render, screen} from '@testing-library/react';
import IssuesComponent from "../IssuesComponent";
import Logo from "../Logo";

/*jest --updateSnapshot NÃ¥r snapshoten endrer seg*/
describe("Snapshot tests", () => {
  it("Logo matches snapshot", () => {
    const comp = renderer.create(<Logo />).toJSON();
    expect(comp).toMatchSnapshot();
  });

  it("Visualisation of Issues matches snapshot", () => {
    //Klarer ikke rendre Responsive container
    const comp = renderer.create(<IssuesComponent />).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
