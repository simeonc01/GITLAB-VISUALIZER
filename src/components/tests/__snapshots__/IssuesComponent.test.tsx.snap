// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot Only component test 1`] = `
<svg
  height="55"
  width="55"
>
  <line
    stroke="white"
    stroke-width="2px"
    x1="25"
    x2="47"
    y1="25"
    y2="25"
  />
  <circle
    cx="25"
    cy="25"
    fill="grey"
    mask="url(#quarter)"
    r="40%"
    stroke="white"
    stroke-width="4%"
  />
  <line
    stroke="white"
    stroke-width="2px"
    x1="25"
    x2="25"
    y1="3"
    y2="25"
  />
  <line
    stroke="white"
    stroke-width="4px"
    x1="25"
    x2="41.5"
    y1="25"
    y2="8.5"
  />
  <defs>
    <mask
      id="quarter"
    >
      <rect
        fill="white"
        height="50"
        width="25"
        x="0"
        y="0"
      />
      <rect
        fill="white"
        height="25"
        width="50"
        x="0"
        y="25"
      />
    </mask>
  </defs>
</svg>
`;
